tensilelite/Tensile/KernelWriter.py

```diff
@@ -2627,33 +2627,59 @@ def kernelBody( self, kernel, tensorParametersA, tensorParametersB ):
       # if swapGlobalRoad is true, swap the order of global read (B->A)
       tensorParameters1st = tensorParametersA
       tensorParameters2nd = tensorParametersB
+      tailLoopOpt1st = kernel["tailLoopOptA"]
+      tailLoopOpt2nd = kernel["tailLoopOptB"]
+
       tc1 = 'A'
       tc2 = 'B'
       if self.isSwapGlobalReadOrderForDtvOrDtl(kernel):
         tensorParameters1st, tensorParameters2nd = tensorParameters2nd, tensorParameters1st
+        tailLoopOpt1st, tailLoopOpt2nd = tailLoopOpt2nd, tailLoopOpt1st
         tc1, tc2 = tc2, tc1
 
       globalReadMode1st = 2 if (((tensorParameters1st["glvw"] * tensorParameters1st["bpeGR"]) < 4) or \
-                               kernel["tailLoopOpt"] == False) else 0
+                               tailLoopOpt1st == False) else 3
       globalReadMode2nd = 2 if (((tensorParameters2nd["glvw"] * tensorParameters2nd["bpeGR"]) < 4) or \
-                               kernel["tailLoopOpt"] == False) else 0
-      globalReadMode1st = 0 if tensorParameters1st["isSwizzled"] else globalReadMode1st
-      globalReadMode2nd = 0 if tensorParameters2nd["isSwizzled"] else globalReadMode2nd
+                               tailLoopOpt2nd == False) else 3
+
+      globalReadMode1st = 3 if tensorParameters1st["isSwizzled"] else globalReadMode1st
+      globalReadMode2nd = 3 if tensorParameters2nd["isSwizzled"] else globalReadMode2nd
```

Comment:
Can you put more comments about what is globalReadMode1st and globalReadMode2nd here.